src/crypto_helper/input/bcrypt.rs

```diff
@@ -0,0 +1,71 @@
+use web_sys::HtmlInputElement;
+use yew::{Properties, Callback, Html, html, function_component, classes, TargetCast};
+use yew_notifications::{use_notification, Notification, NotificationType};
+
+use crate::crypto_helper::algorithm::{BcryptAction, BcryptHashAction, BcryptInput as BI};
+use crate::common::{SwitchProps, Switch};
+
+
+#[derive(PartialEq, Properties, Clone)]
+pub struct BcryptInputProps {
+    pub input: BI,
+    pub bcrypt_input_setter: Callback<BI>,
+}
+
+#[function_component(BcryptInput)]
+pub fn bcrypt_input(
+    input_props: &BcryptInputProps,
+) -> Html {
+    let data = input_props.input.data.clone();
+    let input_setter = input_props.bcrypt_input_setter.clone();
+    let notifications = use_notification::<Notification>();
+    let on_cost_input = Callback::from(move |event: html::oninput::Event| {
+        match event.target_unchecked_into::<HtmlInputElement>().value().parse::<u32>() {
+            Ok(rounds) => input_setter.emit(BI {
+                action: BcryptAction::Hash(BcryptHashAction {
+                    rounds,
+                    salt: Vec::new(), // todo!
+                }),
+                data: data.clone()
+            }),
+            Err(err) => notifications.spawn(Notification::new(
+                NotificationType::Error,
+                "Bcrypt cost parsing error",
+                err.to_string(),
+                Notification::NOTIFICATION_LIFETIME,
+            )),
```

Comment:
instead of spawning notifications, you can paint the input component in red. example: https://github.com/TheBestTvarynka/crypto-helper/blob/main/src/common/byte_input.rs#L70